File: baseble/src/main/java/com/fcbox/ble/baseble/exception/GattException.java
package com.fcbox.ble.baseble.exception;


import com.fcbox.ble.baseble.common.BleExceptionCode;

/**
 * Gatt异常
 */
@SuppressWarnings("unused")
public class GattException extends BleException {
    private int gattStatus;

    public GattException(int gattStatus) {
        super(BleExceptionCode.GATT_ERR, "Gatt Exception Occurred! ");
        this.gattStatus = gattStatus;
    }

    public int getGattStatus() {
        return gattStatus;
    }

    public GattException setGattStatus(int gattStatus) {
        this.gattStatus = gattStatus;
        return this;
    }

    @Override
    public String toString() {
        return "GattException{" +
                "gattStatus=" + gattStatus +
                '}' + super.toString();
    }
}

File: baseble/src/main/java/com/xzy/ble/baseble/exception/GattException.kt
package com.xzy.ble.baseble.exception


import com.xzy.ble.baseble.common.BleExceptionCode

/**
 * Gatt异常
 */
@Suppress("unused")
class GattException(private var gattStatus: Int) : BleException(BleExceptionCode.GATT_ERR, "Gatt Exception Occurred! ") {

    fun getGattStatus(): Int {
        return gattStatus
    }

    fun setGattStatus(gattStatus: Int): GattException {
        this.gattStatus = gattStatus
        return this
    }

    override fun toString(): String {
        return "GattException{" +
                "gattStatus=" + gattStatus +
                '}'.toString() + super.toString()
    }
}
